src/pybind/mgr/volumes/fs/async_cloner.py

```diff
@@ -329,10 +329,14 @@ def cancel_job(self, volname, job):
                         if not track_idx:
                             log.warning("cannot lookup clone tracking index for {0}".format(clone_subvolume.base_path))
                             raise VolumeException(-errno.EINVAL, "error canceling clone")
-                        if SubvolumeOpSm.is_init_state(SubvolumeTypes.TYPE_CLONE, clone_state):
-                            # clone has not started yet -- cancel right away.
-                            self._cancel_pending_clone(fs_handle, clone_subvolume, clonename, groupname, status, track_idx)
-                            return
+                        clone_job = (track_idx, clone_subvolume.base_path)
+                        jobs = [j[0] for j in self.jobs[volname]]
+                        with lock_timeout_log(self.lock):
```

Comment:
I do not recall the order of locking -- global lock vs this lock -- please double check that once.

Reply:
Clone operation mostly open the volume (default with global lock) and then the async_job lock. So we should be good here. The below code at L345 confirms that order too, the volume is being opened without lock after the async_job lock.